services/wallet/collectibles/manager.go

```diff
@@ -420,50 +406,77 @@ func (o *Manager) FetchCollectibleOwnershipByOwner(ctx context.Context, chainID
 // If asyncFetch is true, empty metadata will be returned for any missing collectibles and an EventCollectiblesDataUpdated will be sent when the data is ready.
 // If asyncFetch is false, it will wait for all collectibles' metadata to be retrieved before returning.
 func (o *Manager) FetchAssetsByCollectibleUniqueID(ctx context.Context, uniqueIDs []thirdparty.CollectibleUniqueID, asyncFetch bool) ([]thirdparty.FullCollectibleData, error) {
-	missingIDs, err := o.collectiblesDataDB.GetIDsNotInDB(uniqueIDs)
+	err := o.FetchMissingAssetsByCollectibleUniqueID(ctx, uniqueIDs, asyncFetch)
 	if err != nil {
 		return nil, err
 	}
 
+	return o.getCacheFullCollectibleData(uniqueIDs)
+}
+
+func (o *Manager) FetchMissingAssetsByCollectibleUniqueID(ctx context.Context, uniqueIDs []thirdparty.CollectibleUniqueID, asyncFetch bool) error {
+	missingIDs, err := o.collectiblesDataDB.GetIDsNotInDB(uniqueIDs)
+	if err != nil {
+		return err
+	}
+
 	missingIDsPerChainID := thirdparty.GroupCollectibleUIDsByChainID(missingIDs)
 
-	group := async.NewGroup(ctx)
-	group.Add(func(ctx context.Context) error {
-		for chainID, idsToFetch := range missingIDsPerChainID {
+	// Atomic group stores the error from the first failed command and stops other commands on error
+	group := async.NewAtomicGroup(ctx)
+	for chainID, idsToFetch := range missingIDsPerChainID {
+		group.Add(func(ctx context.Context) error {
 			defer o.checkConnectionStatus(chainID)
 
-			for _, provider := range o.collectibleDataProviders {
-				if !provider.IsChainSupported(chainID) {
-					continue
-				}
+			fetchedAssets, err := o.fetchMissingAssetsForChainByCollectibleUniqueID(ctx, chainID, idsToFetch)
+			if err != nil {
+				log.Error("FetchMissingAssetsByCollectibleUniqueID failed for", "chainID", chainID, "ids", idsToFetch, "err", err)
+				return err
+			}
 
-				fetchedAssets, err := provider.FetchAssetsByCollectibleUniqueID(ctx, idsToFetch)
-				if err != nil {
-					log.Error("FetchAssetsByCollectibleUniqueID failed for", "provider", provider.ID(), "chainID", chainID, "err", err)
-					continue
-				}
+			updatedCollectibles, err := o.processFullCollectibleData(ctx, fetchedAssets, asyncFetch)
+			if err != nil {
+				log.Error("processFullCollectibleData failed for", "chainID", chainID, "len(fetchedAssets)", len(fetchedAssets), "err", err)
+				return err
+			}
 
-				updatedCollectibles, err := o.processFullCollectibleData(ctx, fetchedAssets, asyncFetch)
-				if err != nil {
-					log.Error("processFullCollectibleData failed for", "provider", provider.ID(), "chainID", chainID, "len(fetchedAssets)", len(fetchedAssets), "err", err)
-					return err
-				}
+			o.signalUpdatedCollectiblesData(updatedCollectibles)
+			return nil
+		})
+	}
 
-				if asyncFetch {
-					o.signalUpdatedCollectiblesData(updatedCollectibles)
-				}
-				break
-			}
+	group.Wait()
```

Comment:
hmm I'm gonna need to check if there's a chance this will block for a considerable time, which is not desired when `asyncFetch == true`